### PR TITLE
Added decode support for DDS Dxt10 format

### PR DESCRIPTION
**Pull request by SubhadeepG**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/ImageMagick/ImageMagick/pulls) open
- [x] I have verified that I am following the existing coding patterns and practices as demonstrated in the repository.

### Description
I have added support for decoding the DDS BC7 format. I have also fixed a typo.
Moved the description of ReadDDSImage function to its correct location.
This PR fixes the issue #2451

### CODE REVIEW

File: MagickCore/compress.h [hunk 1]
```diff
@@ -31,6 +31,7 @@ typedef enum
   DXT1Compression,
   DXT3Compression,
   DXT5Compression,
+  BC7Compression,
```
Comment (dlemstra) [inline MagickCore/compress.h:34]: I haven't had time to do a proper review of this PR yet will try to do that this weekend. Already a quick remark about this change. New values can only be added at the end.

File: coders/dds.c [hunk 2]
```diff
@@ -1624,6 +1832,355 @@ static MagickBooleanType ReadDXT5(const ImageInfo *image_info,Image *image,
...
+    return AnchorIndexTable[1][partitionid] == pixelIndex;
+  /* for second subset of three subset partitioning */
+  if (subsetIndex == 1 && numSubsets == 3)
+    return AnchorIndexTable[2][partitionid] == pixelIndex;
+
+  /* for third subset of three subset partitioning */
+  return AnchorIndexTable[3][partitionid] == pixelIndex;
+}
+
+static MagickBooleanType ReadEndpoints(BC7Colors *endpoints,
```
Comment (dlemstra) [inline coders/dds.c:1881]: The return value is always `MagickTrue` and not check. Can we change this to `void` instead?

**Review by dlemstra — COMMENTED** [review]
Thanks for your help with this. I left some comments on your changes. I don't mind doing them myself if you prefer that.

**Comment by codecat** [comment]
This is awesome, thank you for putting the work into this.

As per the original Github issue, I have re-uploaded one of the images I had issues with: [5f62256170bf3.zip](https://github.com/ImageMagick/ImageMagick/files/7110418/5f62256170bf3.zip)

To note, Paint .Net loads this image vertically mirrored, which potentially seems like a bug with PDN rather than the actual image.

File: coders/dds.c [hunk 3]
```diff
@@ -1624,6 +1832,355 @@ static MagickBooleanType ReadDXT5(const ImageInfo *image_info,Image *image,
     return(SkipDXTMipmaps(image,dds_info,16,exception));
 }
 
+static unsigned char GetBit(const unsigned char *block,size_t *startBit)
+{
+  size_t index=(*startBit) >> 3;
+  size_t base=(*startBit) - (index << 3);
+  (*startBit)++;
+  return ((block[index] >> base) & 0x01);
```
[the remark this answers is not in the document]
Comment (SubhadeepG) [reply coders/dds.c:1840]: The value of index can exceed 15 when wrong input is provided but that never happens in my code. For safety I will add a boundary check.

**Comment by SubhadeepG** [comment]
@codecat The image you provided is stored in vertically inverted position. There is nothing wrong with Paint .net.

**Pull request merged**